src/view-resources.js

```diff
@@ -283,4 +288,41 @@ export class ViewResources {
   getValue(name: string): any {
     return this.values[name] || (this.hasParent ? this.parent.getValue(name) : null);
   }
+
+  /**
+   * Auto register a resources based on its metadata or convention
+   * Will fallback to custom element if no metadata found and all conventions fail
+   * @param {Container} container
+   * @param {Function} impl
+   * @returns {HtmlBehaviorResource | ValueConverterResource | BindingBehaviorResource | ViewEngineHooksResource}
+   */
+  autoRegister(container, impl) {
+    let resourceTypeMeta = metadata.get(metadata.resource, impl);
+    if (resourceTypeMeta) {
+      if (resourceTypeMeta instanceof HtmlBehaviorResource) {
+        if (resourceTypeMeta.attributeName === null && resourceTypeMeta.elementName === null) {
+          //no customeElement or customAttribute but behavior added by other metadata
+          HtmlBehaviorResource.convention(impl.name, resourceTypeMeta);
```

Comment:
The only real issue I see if the use of `Function#name` which will not be preserved when minifying code. Is the assumption for this feature that developers will always explicitly provide the decorator with the name so that it never falls back to the `Function#name`?

Reply:
Probably that's the expected behavior, since devs need to be aware of their script minimization, I think. And it's for `UglifyJs`, it's the `keep_classname`, that will do the job. Besides that, I see no harm on enforcing it if devs want to go all in on static code analysis

Reply:
>since devs need to be aware of their script minimization
nobody is aware of nothing (-:
example: https://github.com/aurelia/templating/issues/609#event-1547295242

Reply:
I am aware of that

Reply:
Probably we should treat this as private API first and only used by the config Api like framework configuration or static view. If folks want it, we could open it but view resources is something i would normally touch. We may also need a flag on the typing to tell it is for internal use. About doc, it would be better explained by a real usage of this api like config

Reply:
That makes sense. You can just add a doc comment that it's an internal API, not supported for public use and may change without warning, etc.